src/ngraph/opsets/opset.hpp

```diff
@@ -16,35 +16,87 @@
 
 #pragma once
 
+#include <map>
+#include <mutex>
 #include <set>
 
+#include "ngraph/factory.hpp"
 #include "ngraph/node.hpp"
 
 namespace ngraph
 {
+    /// \brief Run-time opset information
     class OpSet
```

Comment:
suggest to export with `NGRAPH_API`

Let's use this macro for all new / touched classes which are assumed to be plugin ngraph API.